libjulia/optimiser/FullInliner.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Optimiser component that performs function inlining for arbitrary functions.
+ */
+
+#include <libjulia/optimiser/FullInliner.h>
+
+#include <libjulia/optimiser/ASTCopier.h>
+#include <libjulia/optimiser/ASTWalker.h>
+#include <libjulia/optimiser/NameCollector.h>
+
+#include <libsolidity/inlineasm/AsmData.h>
+
+#include <libsolidity/interface/Exceptions.h>
+
+#include <libdevcore/CommonData.h>
+
+#include <boost/range/adaptor/reversed.hpp>
+
+using namespace std;
+using namespace dev;
+using namespace dev::julia;
+using namespace dev::solidity;
+
+
+
+FullInliner::FullInliner(Block& _ast):
+	m_ast(_ast)
+{
+	solAssert(m_ast.statements.size() >= 1, "");
+	m_nameDispenser.m_usedNames = NameCollector(m_ast).names();
+
+	for (size_t i = 1; i < m_ast.statements.size(); ++i)
+	{
+		FunctionDefinition& fun = boost::get<FunctionDefinition>(m_ast.statements.at(i));
```

Comment:
How about adding two asserts (`at(0)` to be a block and `at(i)` to be a `FunctionDefinition`) - that is what the readme is saying?

Reply:
This is done.